test/lib/docker.test.ts

```diff
@@ -66,6 +67,55 @@ test("docker run", async (t) => {
   });
 });
 
+test("save from docker daemon", async (t) => {
+  const targetImage = "openjdk-test:latest";
+  const docker = new Docker(targetImage);
+  await subProcess.execute("docker", [
+    "load",
+    "--input",
+    path.join(
+      __dirname,
+      "../fixtures/docker-archives",
+      "docker-save/openjdk.tar",
+    ),
+  ]);
+
+  t.test("image saved to specified location", async (t) => {
+    const destination = path.join(os.tmpdir(), "image.tar");
+
+    t.tearDown(() => {
+      if (existsSync(destination)) {
+        unlinkSync(destination);
+      }
+    });
+
+    await docker.save(targetImage, destination);
+    t.true(existsSync(destination));
```

Comment:
would it be sensible to assert anything else here beyond the file existing? for example the file size from the fixtures' dir is the same as the new file?

Reply:
Yes, I was was considering doing a checksum comparison between the original tar and the new one.